src/app/components/header/header.component.ts

```diff
@@ -1,10 +1,33 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { Access } from '../../core/models/user.model';
+import { selectCurrentAccess } from '../../core/store/user/user.selectors';
+import { TrainState } from '../../core/store/store.model';
+import { signIn } from '../../core/store/user/user.actions';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrl: './header.component.scss'
+  styleUrl: './header.component.scss',
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
+  role$!: Observable<Access>;
```

Comment:
I think here need to add access modifiers

Reply:
yeah, i think it's will be better solution

Reply:
Done